src/mlpack/methods/fastmks/fastmks_model_impl.hpp

```diff
@@ -34,8 +34,7 @@ void BuildFastMKSModel(FastMKS<KernelType>& f,
     // Create the tree with the specified base.
     Timer::Start("tree_building");
     metric::IPMetric<KernelType> metric(k);
-    typename FastMKS<KernelType>::Tree* tree =
-        new typename FastMKS<KernelType>::Tree(referenceData, metric, base);
+    typename FastMKS<KernelType>::Tree tree(referenceData, metric, base);
     Timer::Stop("tree_building");
 
     f.Train(tree);
```

Comment:
We should use `std::move(tree)` here to avoid copying the entire tree---that will be a big speed improvement.